src/components/Video/Video.js

```diff
@@ -17,52 +12,71 @@ class Video extends CloudinaryComponent {
+      children,
+      ...options
+    } = this.getMergedProps();
+
     sourceTypes = sourceTypes || Cloudinary.DEFAULT_VIDEO_PARAMS.source_types;
     options = CloudinaryComponent.normalizeOptions(options, {});
-    let cld = Cloudinary.new(Util.withSnakeCaseKeys(options));
-    let sources = [];
-    let tagAttributes = Transformation.new(options).toHtmlAttributes();
-    let childTransformations = this.getTransformation(options);
-    if (Util.isPlainObject(poster)) {
-      let defaults = poster.publicId !== undefined && poster.publicId !== null ? Cloudinary.DEFAULT_IMAGE_PARAMS : DEFAULT_POSTER_OPTIONS;
-      poster = cld.url(poster.publicId || publicId, Util.defaults({}, Util.withSnakeCaseKeys(poster), defaults));
-    }
-    if (!Util.isEmpty(poster)) {
-      tagAttributes.poster = poster;
-    }
-    if (!Util.isEmpty(this.state.poster)) {
-      tagAttributes.poster = this.state.poster;
-    }
 
+    const childTransformations = this.getTransformation({...options, children});
+    const snakeCaseOptions = Util.withSnakeCaseKeys(options);
+    const cld = Cloudinary.new(snakeCaseOptions);
+    const tagAttributes = cld.videoTag(publicId, snakeCaseOptions).attributes();
+
+    let sources = null;
     if (Util.isArray(sourceTypes)) {
-      sources = sourceTypes.map(srcType => {
-          let sourceTransformation = sourceTransformations[srcType] || {};
-          let src = cld.url(publicId, Util.defaults({}, sourceTransformation, childTransformations, {resource_type: 'video', format: srcType}));
-          let mimeType = 'video/' + (srcType === 'ogv' ? 'ogg' : srcType);
+      sources = sourceTypes.map(sourceType => {
+          const urlOptions = this.getSourceUrlOptions(childTransformations, sourceTransformation, sourceType);
+          const src = cld.url(publicId, urlOptions);
+          const mimeType = 'video/' + (sourceType === 'ogv' ? 'ogg' : sourceType);
           return <source key={mimeType} src={src} type={mimeType}/>;
         }
       );
     } else {
-      let sourceTransformation = sourceTransformations[sourceTypes] || {};
-      tagAttributes.src = cld.url(publicId, Util.defaults({}, sourceTransformation, childTransformations, {resource_type: 'video', format: sourceTypes}));
+      const urlOptions = this.getSourceUrlOptions(childTransformations, sourceTransformation, sourceTypes);
+      tagAttributes.src = cld.url(publicId, urlOptions);
     }
 
+    return {sources, tagAttributes};
+  };
+
+  render() {
+    const {innerRef, fallback, children} = this.props;
+    const {tagAttributes, sources} = this.getVideoTagProps();
+
     return (
-      <video ref={innerRef} {...tagAttributes}>
+      <video
+        ref={innerRef}
+        {...tagAttributes}>
         {sources}
         {fallback}
-        {this.props.children}
+        {children}
       </video>
     );
   }
 }
+
 Video.propTypes = {publicId: PropTypes.string};
 Video.defaultProps = {};
-Video.contextTypes = CloudinaryComponent.contextTypes;
 
```

Comment:
@maoznir , so... no 'poster' handling at all? it's all handled by core? 
Are all poster options covered by tests?

Reply:
all attributes except for source handled by core + added tests.
This way we get an alignment with angular sdk, except in react sdk's video component we don't manipulate the dom directly